mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -406,6 +435,14 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
 
   private[regression] val epsilon: Double = 1E-16
 
+  /** Checks whether weight column is set and nonempty */
+  private[regression] def isSetWeightCol(params: GeneralizedLinearRegressionBase): Boolean =
+    params.isSet(params.weightCol) && !params.getWeightCol.isEmpty
+
+  /** Checks whether offset column is set and nonempty */
+  private[regression] def isSetOffsetCol(params: GeneralizedLinearRegressionBase): Boolean =
+    params.isSet(params.offsetCol) && !params.getOffsetCol.isEmpty
```

Comment:
I adopted `params.getOffsetCol.nonEmpty`. 
As for `isDefined` VS `isSet`, in my other PR #16344, @yanboliang suggested `isSet` is more accurate and we should be using `isSet` and may need to change the existing `isDefined` at some point. For this reason, I'm using `isSet` (which I believe checks whether the param is explicitly set by the user).